docs/source/en/model_doc/seamless_m4t.md

```diff
@@ -0,0 +1,202 @@
+
+## Usage
+
+First, load the processor and a checkpoint of the model:
+
+```python
+>>> from transformers import AutoProcessor, SeamlessM4TModel
+
+>>> processor = AutoProcessor.from_pretrained("ylacombe/hf-seamless-m4t-medium")
+>>> model = SeamlessM4TModel.from_pretrained("ylacombe/hf-seamless-m4t-medium")
+```
+
+You can seamlessly use this model on text or on audio, to generated either translated text or translated audio.
+
+### Speech
+
+You can easily generate translated speech with [`SeamlessM4TModel.generate`]. Here is an example showing how to generate speech from English to Russian.
+
+```python
+>>> inputs = processor(text = "Hello, my dog is cute", src_lang="eng", return_tensors="pt")
+
+>>> audio_array = model.generate(**inputs, tgt_lang="rus")
+>>> audio_array = audio_array[0].cpu().numpy().squeeze()
+```
+
+You can also translate directly from a speech waveform. Here is an example from Arabic to English:
+
+```python
+>>> from datasets import load_dataset
+
+>>> dataset = load_dataset("arabic_speech_corpus", split="test[0:1]")
+
+>>> audio_sample = dataset["audio"][0]["array"]
+ 
+>>> inputs = processor(audios = audio_sample, return_tensors="pt")
+
+>>> audio_array = model.generate(**inputs, tgt_lang="rus")
+>>> audio_array = audio_array[0].cpu().numpy().squeeze()
+```
+
+#### Tips
```

Comment:
I would advocate for using the top-level `SeamlessM4TModel` for all the tasks in this doc, and then highlighting that the individual models can be used once at the end in the `tips` section, maybe with one contrasting example